src/test_helper.rs

```diff
@@ -190,3 +201,33 @@ impl Widget for DebugWidget {
         self.content.lifecycle(cx, event);
     }
 }
+
+/// Utility for visual snapshot test debugging
+///
+/// If the environment variable `BEBUG_SNAPSHOT` is set when tests are run, the terminal buffer is
+/// dumped to stdout.
+///
+/// ```sh
+/// DEBUG_SNAPSHOT=1 cargo test --lib -- --nocapture --test simple_block_test
+/// ```
+///
+/// !!! The normal test output frequently interferes which results in scrambled output, especially
+/// when multiple tests are run at once.
+/// Running it multiple times might usually leads to good output (for now, with small widget output)
+pub fn print_buffer(buffer: &Buffer) -> std::io::Result<()> {
+    if env::var("DEBUG_SNAPSHOT").is_ok() {
```

Comment:
Nice job on the doc comment.

I tried the one test with `.with_borders.with_borders(BorderKind::Rounded)` and it seems to put the whole buffer linewise on one line to stdout, do you know whether it's possible to fix this?

Reply:
That was the sole purpose, but I don't know much about Ratatui and Crossterm yet and I failed to try a more "elaborate" 😉 example with multiple lines.

I will of course make an attempt to get that working.

Reply:
I will call `cargo outdated` frequently to check anyway and if it doesn't bother you just upgrade when newer versions become available.

Reply:
Anyway I don't think printing the buffer is super important (that doesn't mean I'm against merging this), as the diff of insta is already quite helpful I think (which seems to correctly display the full buffer minus styling).

You could split the Windows CI fix into a separate PR though.

Reply:
Snapshotting the buffer includes styles but in a section below the text. So if it unintentionally changes, the test catches it. It's just a bit cumbersome to manually verify whether it's correct. 

Printing the buffer was only meant as interactive debugging tool.
I think merging should be fine. It reduces potential merge conflicts. 
I'll iterate on it and if I don't get it working properly I'll dump it.

Reply:
> Printing the buffer was only meant as interactive debugging tool.

FWIW I'm using the examples for this (as it directly serves a documentation purpose as well). Long-term automated tests are certainly more robust though...

> I think merging should be fine. It reduces potential merge conflicts.

Fine by me, doesn't really introduce much code to maintain anyway.